src/Quidjibo.DataProtection/Protectors/AesPayloadProtector.cs

```diff
@@ -22,11 +27,12 @@ public AesPayloadProtector(IKeyProvider keyProvider)
         /*
          * Encrypted payload:
          * 
-         * |--- IV (16 bytes) ---|--- MAC (32 bytes) ---|--- Ciphertext (n bytes) ---|
+         * |--- Version (1 byte) ---|--- IV (16 bytes) ---|--- MAC (32 bytes) ---|--- Ciphertext (n bytes) ---|
          * 
          */
         public async Task<byte[]> ProtectAsync(byte[] payload, CancellationToken cancellationToken)
         {
+            var version = new ArraySegment<byte>(_version1);
```

Comment:
ArraySegment is kind of icky but works well enough for netstandard2.0. This can be greatly improved in the future on netstandard2.1 since that is when many Span APIs will become available and we can switch to Span / Memory.